File: java-tests/src/main/java/com/benjaminsproule/swagger/gradleplugin/test/java/jaxrs/TestResourceWithClassAnnotation.java
package com.benjaminsproule.swagger.gradleplugin.test.java.jaxrs;

import com.benjaminsproule.swagger.gradleplugin.test.model.*;
import com.fasterxml.jackson.annotation.JsonView;
import io.swagger.annotations.*;

import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import java.util.List;

import static java.util.Collections.singletonList;

@Api(tags = "Test", description = "Test resource", authorizations = {@Authorization("basic")})
@Path("/root/withannotation")
public class TestResourceWithClassAnnotation {

    @ApiOperation("A basic operation")
    @Path("/basic")
    @GET
    public String basic() {
        return "";
    }

    @ApiOperation("A default operation")
    @Path("/default")
    @GET
    public Response defaultResponse() {
        return Response.ok().build();
    }

    @ApiOperation("A generics operation")
    @Path("/generics")
    @POST
    public List<String> generics(@ApiParam List<RequestModel> body) {
        return singletonList("");
    }

    @ApiOperation("Consumes and Produces operation")
    @Path("/datatype")
    @Consumes("application/json")
    @Produces("application/json")
    @POST
    public Response dataType(@ApiParam RequestModel requestModel) {
        return Response.ok().build();
    }

    @ApiOperation(value = "A response operation", response = ResponseModel.class)
    @Path("/response")
    @POST
    public ResponseModel response() {
        return new ResponseModel();
    }

    @ApiOperation(value = "A response container operation", response = ResponseModel.class, responseContainer = "List")
    @Path("/responseContainer")
    @POST
    public List<ResponseModel> responseContainer() {
        return singletonList(new ResponseModel());
    }

    @ApiOperation("An extended operation")
    @Path("/extended")
    @GET
    public SubResponseModel extended() {
        return new SubResponseModel();
    }

    @ApiOperation("A deprecated operation")
    @Path("/deprecated")
    @GET
    @Deprecated
    public String deprecated() {
        return "";
    }

    @ApiOperation(value = "An auth operation", authorizations = {
        @Authorization(value = "oauth2", scopes = {
            @AuthorizationScope(scope = "scope", description = "scope description")
        })
    })
    @Path("/auth")
    @GET
    public String withAuth() {
        return "";
    }

    @ApiOperation("A model operation")
    @Path("/model")
    @GET
    public String model() {
        return "";
    }

    @ApiOperation("An overriden operation")
    @Path("/overriden")
    @GET
    public String overriden() {
        return "";
    }

    @ApiOperation("An overriden operation")
    @Path("/overridenWithoutDescription")
    @GET
    public String overridenWithoutDescription() {
        return "";
    }

    @ApiOperation(value = "A hidden operation", hidden = true)
    @Path("/hidden")
    @GET
    public String hidden() {
        return "";
    }

    @ApiOperation("A multiple parameters operation")
    @Path("/multipleParameters/{parameter1}")
    @GET
    public String multipleParameters(@PathParam("parameter1") Double parameterDouble, @QueryParam("parameter2") Boolean parameterBool) {
        return "";
    }

    String ignoredModel(IgnoredModel ignoredModel) {
        return "";
    }

    @ApiOperation("A PATCH operation")
    @Path("/patch")
    @PATCH
    public String patch() {
        return "";
    }

    @ApiOperation("An OPTIONS operation")
    @Path("/options")
    @OPTIONS
    public Response options() {
        return Response.ok().build();
    }

    @ApiOperation("An HEAD operation")
    @Path("/head")
    @HEAD
    public String head() {
        return "";
    }

    @ApiOperation(value = "An implicit params operation")
    @ApiImplicitParams({
        @ApiImplicitParam(name = "body", required = true, dataType = "com.benjaminsproule.swagger.gradleplugin.test.model.RequestModel", paramType = "body"),
        @ApiImplicitParam(name = "id", value = "Implicit parameter of primitive type string", dataType = "string", paramType = "header")
    })
    @Path("/implicitparams")
    @POST
    public String implicitParams(String requestModel) {
        return "";
    }

    @ApiOperation(value = "A created request operation", code = 201)
    @Path("/createdrequest")
    @POST
    public String createdRequest() {
        return "";
    }

    @ApiResponses(
        value = {
            @ApiResponse(code = 201, message = "Success", response = String.class),
            @ApiResponse(code = 422, message = "Business errors", response = String.class)}
    )
    @Path("/apiresponses")
    @POST
    public String apiResponses() {
        return "";
    }

    @ApiOperation(value = "A inner JSON sub type operation")
    @Path("/innerjsonsubtype")
    @GET
    public OuterJsonSubType innerJsonSubType() {
        return new OuterJsonSubType();
    }

    @ApiOperation("With JsonViewOne specification")
    @Path("/root/withoutannotation/withjsonview1")
    @GET
    @JsonView(TestJsonViewOne.class)
    public TestJsonViewEntity withJsonViewOne() {
        return null;
    }

    @ApiOperation("With JsonViewTwo specification")
    @Path("/root/withoutannotation/withjsonview2")
    @GET
    @JsonView(TestJsonViewTwo.class)
    public TestJsonViewEntity withJsonViewTwo() {
        return null;
    }

    @ApiOperation("Entity definition has to contain all possible fields")
    @Path("/root/withoutannotation/withoutjsonview")
    @GET
    public TestJsonViewEntity withoutJsonView() {
        return null;
    }
}

File: scala-tests/src/main/scala/com/benjaminsproule/swagger/gradleplugin/test/scala/jaxrs/TestResourceWithClassAnnotation.scala
package com.benjaminsproule.swagger.gradleplugin.test.scala.jaxrs

import com.benjaminsproule.swagger.gradleplugin.test.model._
import com.fasterxml.jackson.annotation.JsonView
import io.swagger.annotations._
import javax.ws.rs._
import javax.ws.rs.core.Response

@Api(tags = Array("Test"), description = "Test resource", authorizations = Array(
    new Authorization("basic")
))
@Path("/root/withannotation")
class TestResourceWithClassAnnotation {

    @ApiOperation("A basic operation")
    @Path("/basic")
    @GET
    def basic(): String = {
        ""
    }

    @ApiOperation("A default operation")
    @Path("/default")
    @GET
    def defaultResponse(): Response = {
        Response.ok().build()
    }

    @ApiOperation("A generics operation")
    @Path("/generics")
    @POST
    def generics(@ApiParam body: List[RequestModel]): List[String] = {
        List("")
    }

    @ApiOperation("Consumes and Produces operation")
    @Path("/datatype")
    @Consumes(Array("application/json"))
    @Produces(Array("application/json"))
    @POST
    def dataType(@ApiParam requestModel: RequestModel): Response = {
        Response.ok().build()
    }

    @ApiOperation(value = "A response operation", response = classOf[ResponseModel])
    @Path("/response")
    @POST
    def response(): ResponseModel = {
        new ResponseModel()
    }

    @ApiOperation(value = "A response container operation", response = classOf[ResponseModel], responseContainer = "List")
    @Path("/responseContainer")
    @POST
    def responseContainer(): List[ResponseModel] = {
        List(new ResponseModel())
    }

    @ApiOperation("An extended operation")
    @Path("/extended")
    @GET
    def extended(): SubResponseModel = {
        new SubResponseModel()
    }

    @ApiOperation("A deprecated operation")
    @Path("/deprecated")
    @GET
    @Deprecated
    def deprecated(): String = {
        ""
    }

    @ApiOperation(value = "An auth operation", authorizations = Array(
        new Authorization(value = "oauth2", scopes = Array(
            new AuthorizationScope(scope = "scope", description = "scope description")
        ))
    ))
    @Path("/auth")
    @GET
    def withAuth(): String = {
        ""
    }

    @ApiOperation("A model operation")
    @Path("/model")
    @GET
    def model(): String = {
        ""
    }

    @ApiOperation("An overriden operation")
    @Path("/overriden")
    @GET
    def overriden(): String = {
        ""
    }

    @ApiOperation("An overriden operation")
    @Path("/overridenWithoutDescription")
    @GET
    def overridenWithoutDescription(): String = {
        ""
    }

    @ApiOperation(value = "A hidden operation", hidden = true)
    @Path("/hidden")
    @GET
    def hidden(): String = {
        ""
    }

    @ApiOperation("A multiple parameters operation")
    @Path("/multipleParameters/{parameter1}")
    @GET
    def multipleParameters(@PathParam("parameter1") parameterDouble: Double, @QueryParam("parameter2") parameterBool: Boolean): String = {
        ""
    }

    def ignoredModel(ignoredModel: IgnoredModel): String = {
        ""
    }

    @ApiOperation("A PATCH operation")
    @Path("/patch")
    @PATCH
    def patch(): String = {
        ""
    }

    @ApiOperation("An OPTIONS operation")
    @Path("/options")
    @OPTIONS
    def options(): Response = {
        Response.ok().build()
    }

    @ApiOperation("An HEAD operation")
    @Path("/head")
    @HEAD
    def head(): String = {
        ""
    }

    @ApiOperation(value = "An implicit params operation")
    @ApiImplicitParams(Array(
        new ApiImplicitParam(name = "body", required = true, dataType = "com.benjaminsproule.swagger.gradleplugin.test.model.RequestModel", paramType = "body"),
        new ApiImplicitParam(name = "id", value = "Implicit parameter of primitive type string", dataType = "string", paramType = "header")
    ))
    @Path("/implicitparams")
    @POST
    def implicitParams(requestModel: String): String = {
        ""
    }

    @ApiOperation(value = "A created request operation", code = 201)
    @Path("/createdrequest")
    @POST
    def createdRequest(): String = {
        ""
    }

    @ApiResponses(
        value = Array(
            new ApiResponse(code = 201, message = "Success", response = classOf[String]),
            new ApiResponse(code = 422, message = "Business errors", response = classOf[String])
        )
    )
    @Path("/apiresponses")
    @POST
    def apiResponses(): String = {
        ""
    }

    @ApiOperation(value = "A inner JSON sub type operation")
    @Path("/innerjsonsubtype")
    @GET
    def innerJsonSubType(): OuterJsonSubType = {
        new OuterJsonSubType()
    }

    @ApiOperation(value = "With JsonViewOne specification")
    @Path("/withjsonview1")
    @GET
    @JsonView(value = Array(classOf[TestJsonViewOne]))
    def withJsonViewOne(): TestJsonViewEntity = {
        null
    }

    @ApiOperation("With JsonViewTwo specification")
    @Path("/withjsonview2")
    @GET
    @JsonView(value = Array(classOf[TestJsonViewTwo]))
    def withJsonViewTwo(): TestJsonViewEntity = {
        null
    }

    @ApiOperation("Entity definition has to contain all possible fields")
    @Path("/withoutjsonview1")
    @GET
    def withoutJsonView(): TestJsonViewEntity = {
        null
    }
}
